pkg/storaged/crypto-tab.jsx

```diff
@@ -54,6 +55,33 @@ function parse_tag_mtime(tag) {
         return null;
 }
 
+export function edit_config(block, modify) {
+    var old_config, new_config;
+
+    function commit() {
+        new_config[1]["track-parents"] = { t: 'b', v: true };
+        if (old_config)
+            return block.UpdateConfigurationItem(old_config, new_config, { });
+        else
+            return block.AddConfigurationItem(new_config, { });
+    }
+
+    return block.GetSecretConfiguration({}).then(
+        function (items) {
+            old_config = array_find(items, function (c) { return c[0] == "crypttab" });
+            new_config = ["crypttab", old_config ? Object.assign({ }, old_config[1]) : { }];
```

Comment:
The weird `[ [ "crypttab", { ... } ], ... ]` data structure comes straight from the UDisks2 D-Bus API. 
`new_config` is a shallow copy of `old_config`, or a new empty config if there is no old config.

It would be worthwhile the destructure configs in one place somewhere in the client (including parsing the options string etc) but I was leaving that for later.